### DISCUSSION TITLE
How to configure serverHandlers?

### DISCUSSION THREAD

**Discussion by serhez** [nuxt/nuxt#17415] [Questions]
There is little info in the docs about how to configure `serverHandlers`. I've only found [this](https://v3.nuxtjs.org/api/configuration/nuxt.config#serverhandlers), where it says it is an array (an array of what?). I've also tried looking at the Nitro docs but I found nothing.

 I'm trying to create server middleware which only executes on certain pages, or even on all pages except some. Is this possible? Can I give `serverHandlers` an object with a `route` and `handler` fields? Can the `route` field be a regex so that I can exclude some routes instead of including them?

Thanks!

**Comment by Diizzayy** [comment]
> There is little info in the docs about how to configure serverHandlers

@serhez You can find the documentation here: [Nuxt Server Routes](http://v3.nuxtjs.org/guide/features/server-routes#server-middleware)

> I'm trying to create server middleware which only executes on certain pages, or even on all pages except some. Is this possible? Can I give serverHandlers an object with a route and handler fields? Can the route field be a regex so that I can exclude some routes instead of including them?

It isn't currently possible to configure in that manner, the solution would be to check the request url within your middleware so that your logic is only ran when your conditions ( specific routes, etc ) are met.

Here is an example of how you might go about doing so:

`nuxt-app/server/middleware/demo.ts`
```ts
export default defineEventHandler((event) => {
  const exclusions = ['/one', '/two']

  // check if request url should be excluded from middleware
  if (exclusions.includes(event.req.url)) return
  
  // run middleware logic
})
```

**Reply by serhez** [reply]
Thanks! It's unfortunate that it is not configurable per route ATM, would be a great feature.
